lib/parsers/srtEntries.js

```diff
@@ -1,93 +1,105 @@
-
-  const isIndex = /^\d+$/;
-  const isTimecode = /-->/;
-
-  const validEntries = [];
-  const status = {
-    success: true,
-    invalidEntries: [],
-    invalidTimecodes: [],
+/**
+ * Main function of srtEntries.js
+ */
+function parseSrtEntries(subtitleText, options = { invalidEntries: true, invalidTimecodes: true }) {
+  // Initalizing result object
+  const result = {
+    validEntries: [],
+    status: {
+      success: true,
+      invalidEntries: [],
+      invalidTimecodes: [],
+      invalidIndices: [],
+    },
   };
 
-  const entries = buffered.split(/\n\n/g);
-  entries.forEach(entry => {
-    const trimmed = entry.trim();
-    if (!trimmed) return; // if no value continue
-
-    if (validEntry.test(trimmed)) {
-      const valid = trimmed.split('\n');
-      const timecode = valid[1];
-      const times = timecode.split(validTimecode);
-
-      validEntries.push({
-        id: valid[0],
-        timecode,
-        startMicro: timecodeToMicroseconds(times[1]),
-        endMicro: timecodeToMicroseconds(times[2]),
-        text: valid.slice(2).join('\n'),
-      });
-    } else {
-      // invalid
-      const invalid = trimmed.split('\n');
-
-      let id;
-      let timecode;
-      const text = [];
-      invalid.forEach(line => {
-        if (isIndex.test(line)) id = line;
-        else if (isTimecode.test(line)) timecode = line;
-        else text.push(line);
-      });
-
-      // remove valid entry with missing text
-      if (isIndex.test(id) && validTimecode.test(timecode)) return;
-
-      if (options.invalidTimecodes && !validTimecode.test(timecode)) {
-        status.invalidTimecodes.push({
-          id,
-          timecode,
-        });
-      }
-
-      if (options.invalidEntries) {
-        status.invalidEntries.push({
-          id,
-          timecode,
-          text: text.join('\n'),
-        });
-      }
-
-      status.success = false;
+  // Standardizing line breaks from all OS -  MAC: '\r', UNIX: '\n', WIN '\r\n'
+  subtitleText = subtitleText.replace(/(\r\n|\r)/g, '\n');
+
+  // Creating an array with all potential entries
+  const potentialBlocksArray = subtitleText.match(potentialSrtBlockRegex);
+
+  if (!potentialBlocksArray) {
+    result.validEntries.push('No entries found.');
+    result.status.success = false;
+    return result;
+  }
+
+  // If there is a first entry without timecode that is not an index
+  const invalidFirstEntry = subtitleText.match(invalidFirstEntryRegex)[0].trim();
+  if (invalidFirstEntry.length > 0) {
+    result.status.invalidEntries.push(invalidFirstEntry);
+    result.status.success = false;
+  }
+
+  // Validating all entries and creating return objects
+  const { validEntries, status } = potentialBlocksArray.reduce((acc, cur) => {
+    // Remove excessive newlines
+    cur = cur.replace(/\n{2,}/g, '\n').trim();
+
+    // Check if index exist and valid
+    const potentialIndex = cur.match(potentialIndexRegex);
+    if (potentialIndex && !(/\d+/.test(potentialIndex))) {
```

Comment:
Do we need to be stricter when identifying a valid index?  Should the line be only digits? `/^\d+/$`